holidays/countries/south_korea.py

```diff
@@ -139,7 +139,11 @@ def _populate(self, year):
             self._add_holiday("Tree Planting Day", APR, 5)
 
         # Birthday of the Buddha
-        self._add_chinese_birthday_of_buddha("Birthday of the Buddha")
+        name = "Birthday of the Buddha"
+        buddha_birthday_date = self._add_chinese_birthday_of_buddha(name)
+        self._add_with_alt_holiday(
+            name, buddha_birthday_date, add_hol=False, since=2023
+        )
```

Comment:
For consistency also:
```suggestion
        self._add_with_alt_holiday(
            name,
            self._add_chinese_birthday_of_buddha(name),
            add_hol=False,
            since=2023,
        )
```

Reply:
I noticed that i just passively followed suggestions and missed the consistency. Could i roll back this suggesstion? https://github.com/dr-prodigy/python-holidays/pull/1249#discussion_r1211026735

Because `dec_25` variable name is using in other code(austrailia, boswana, etc). and if using some `_add_...` method in south_korea.py, there is a named variable. like this. 
```
# line:117
name = "Lunar New Year's Day"
new_year_date = self._add_chinese_new_years_day(name)
self._add_chinese_new_years_eve(preceding_day % name)
# line:172
name = "Chuseok"
chuseok_date = self._add_mid_autumn_festival(name)
self._add_holiday(preceding_day % name, chuseok_date + td(days=-1))
```
so i think this code is better.
```
dec_25 = self._add_christmas_day(name)
self._add_with_alt_holiday(name, dec_25, add_hol=False, since=2023)
```
if so, I will roll back the code about Christmas, and do not change about Buddha's birthday. im sorry for bad commit history.

Reply:
> Because `dec_25` variable name is using in other code(austrailia, boswana, etc). and if using some `_add_...` method in south_korea.py, there is a named variable. like this.

Typically, we use named variables when some value is used more than once (as for `chuseok_date`) or action sequence is important (as for `new_year_date`). In other cases, it's better to simplify the expression (as in this case).